src/aiidalab_qe/common/bandpdoswidget.py

```diff
@@ -0,0 +1,833 @@
+                b64_str = base64.b64encode(json_str.encode()).decode()
+                self._download(payload=b64_str, filename=file_name_dos)
+
+        self.pdos_options = ipw.VBox(
+            [
+                self.description,
+                self.dos_atoms_group,
+                self.dos_plot_group,
+                ipw.HBox([self.selected_atoms, self.wrong_syntax]),
+                self.update_plot_button,
+            ]
+        )
+
+        self._initial_view()
+
+        # Set the event handlers
+        self.download_button.on_click(download_data)
+        self.update_plot_button.on_click(self._update_plot)
+
+        super().__init__(
+            children=[
+                self.pdos_options_out,
+                self.download_button,
+                self.bands_widget,  # Add the output widget to the VBox
+            ],
+            **kwargs,
+        )
+        if self.pdos:
+            with self.pdos_options_out:
+                display(self.pdos_options)
+
+    @staticmethod
+    def _download(payload, filename):
+        """Download payload as a file named as filename."""
+        from IPython.display import Javascript
+
+        javas = Javascript(
+            """
+            var link = document.createElement('a');
+            link.href = 'data:text/json;charset=utf-8;base64,{payload}'
+            link.download = "{filename}"
+            document.body.appendChild(link);
+            link.click();
+            document.body.removeChild(link);
+            """.format(
+                payload=payload, filename=filename
+            )
+        )
+        display(javas)
+
+    def _get_dos_data(self):
+        if not self.pdos:
+            return None
+        expanded_selection, syntax_ok = string_range_to_list(
+            self.selected_atoms.value, shift=-1
+        )
+        if syntax_ok:
+            dos = get_pdos_data(
+                self.pdos,
+                group_tag=self.dos_atoms_group.value,
+                plot_tag=self.dos_plot_group.value,
+                selected_atoms=expanded_selection,
+            )
+            return dos
+        else:
+            return None
+
+    def _get_fermi_energy(self):
+        fermi_energy = (
+            self.dos_data["fermi_energy"]
+            if self.dos_data
+            else self.bands.band_parameters["fermi_energy"]
+        )
+        return fermi_energy
+
+    def _get_bands_data(self):
+        if not self.bands:
+            return None
+
+        bands = export_bands_data(self.bands, self.fermi_energy)
+        return bands
+
+    def _bands_labeling(self, bands):
+        """Function to return two lists containing the labels and values (kpoint) for plotting.
+        params:
+        - bands: dictionary from export_bands_data function
+        output: label (list of str), label_values (list of float)
+        """
+        if not bands:
+            return None
+
+        my_paths = bands[0].get("paths")
+        my_labels = []
```

Comment:
```suggestion
        paths = bands[0].get("paths")
        labels = []
```
"my"? The paths and labels have their own free will, they belong to nobody ;P